ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -233,6 +233,9 @@ func (r *RayClusterReconciler) rayClusterReconcile(request ctrl.Request, instanc
 			r.Log.Error(err, "Update status error", "cluster name", request.Name)
 		}
 	}
+	if instance.Status.State == rayiov1alpha1.Pending {
+		return ctrl.Result{RequeueAfter: DefaultRequeueDuration}, nil
+	}
 
 	return ctrl.Result{}, nil
```

Comment:
We should always requeue. I didn't realize we didn't do that. That's very bad. Reconciliation for any controller should happen periodically even if there's isn't a trigger that demands it.

The correct fix for this issue is to change the "successful return" from 
```
return ctrl.Result{}, nil
```
to 
```
return ctrl.Result{RequeueAfter: DefaultRequeueDuration}, nil
```

Reply:
Oh thanks, that's very good to know. Let me update the PR

Reply:
This is a good point.